### PR TITLE
Ensure wave driver and sample programs build in CI, fixing issues turned up during the process.

### LINKED ISSUE #48: Samples should be built by CI
This will keep the sample programs healthy over time. They don't seem to be built by Travis. If the recent fixes work they can maybe be kept buildable this way.

### PR DESCRIPTION
**Pull request by jefftrull**
Display a Travis build status badge and some minimal description in a README

If merged this will resolve #48.

### CODE REVIEW

File: samples/token_statistics/build/Jamfile.v2 [hunk 1]
```diff
@@ -19,7 +19,7 @@ exe token_statistics
     :
         $(SOURCES)
         /boost/wave//boost_wave
-        /boost/program_options//boost_program_options
+        /boost/program_options//boost_program_options/<link>static
```
Comment (hkaiser) [inline samples/token_statistics/build/Jamfile.v2]: What's the rationale of this change? Could you elaborate, please?

Comment (jefftrull) [reply samples/token_statistics/build/Jamfile.v2]: Sure! This stems from a problem linking on MSVC-12. A nearly identical issue was reported at https://github.com/boostorg/program_options/issues/58; after investigating it for a while I realized that the tactic used [here](https://github.com/boostorg/wave/blob/18f834ac355d65d18619af673122c23623e7c912/test/build/Jamfile.v2#L93) would fix this as well.

I did also try adding `BOOST_ALL_DYN_LINK` but then it seemed like some inappropriate things got marked `__dllimport`, in particular `expression_grammar_gen<TokenT>::evaluate` and MSVC didn't like that either.

I haven't tried using `BOOST_PROGRAM_OPTIONS_DYN_LINK` but could attempt it if you prefer that approach.

Comment (hkaiser) [reply samples/token_statistics/build/Jamfile.v2]: Don't worry about this too much, I was just curious.
